src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -355,13 +361,14 @@
                           (utils/multi-swap! state (assoc :loaded-config (get-parsed-response))
                                                    (dissoc :redacted?)))
                       (swap! state assoc :error-response (get-parsed-response false))))})))
+
    :-load-validated-method-config
    (fn [{:keys [state props]}]
      (endpoints/call-ajax-orch
       {:endpoint (endpoints/get-validated-workspace-method-config (:workspace-id props) (:config-id props))
        :on-done (fn [{:keys [success? get-parsed-response status-text]}]
                   (if success?
-                    (let [response (get-parsed-response)
+                    (let [response (fix-validated-method-config (get-parsed-response))
                           fake-inputs-outputs (fn [data]
                                                 (let [method-config (:methodConfiguration data)]
                                                   {:inputs (mapv (fn [k] {:name (name k)}) (keys (:inputs method-config)))
```

Comment:
I would remove the all to get-inputs-outputs if we are now validating against the WDL in the validate endpoint. 
I think this would be accomplished by replacing `(endpoints/call-ajax-orch {...})` with the contents of `fake-inputs-outputs`(which, now, are no longer fake).

Reply:
there's a bit of a behaviour change here that i want to think through.

if the underlying method doesn't exist, your call to `get-validated-workspace-method-config` will now return 404 and not the contents of the MC. so the code as-is will fall through to displaying simply that error and not the MC page at all.

this makes me think that we should first call the _unvalidated_ version of the MC so we have anything to display at all, and then use `get-validated-workspace-method-config` to a) determine if the MC exists, in which case we set `:redacted? true`, and b) to fill in the `:inputs-outputs` for the method itself.

does this make sense to you?

Reply:
That does make sense - but if `get-validated-workspace-method-config succeeds`, we don't need to make the extra call. So, maybe instead of calling the unvalidated version first, we call `get-validated-workspace-method-config` first, and then only if that call fails do we get the unvalidated version.

Reply:
I thought about it some more.

If we call get-unvalidated-MC and get a 404, the MC doesn't exist and we display a page 404 like normal.

If we get an MC, we can then call get-validated-MC. If _that_ returns a 404, the underlying method doesn't exist, so we should set `redacted = true`. We also have to fake the underlying inputs and outputs, much like `fake-inputs-outputs`.

If it returns successful, we still need to infer the underlying method's inputs and outputs, which is a function _like_ `fake-inputs-outputs` but is a bit more involved, because it uses the `missingInputs` field (and others) from the ValidatedMethodConfiguration response.

I think this is overcomplicated compared to just calling `get-inputs-outputs`. Which I think means the call order should be:

1. get-unvalidated-MC (404 = no MC)
2. get-inputs-outputs (404 = no method in which case stop here, use fake-inputs-outputs, add a message saying "we're not validating your expressions until you give us a valid MC")
3. get-validated-MC (use to get real validation errors)

The existence of that message at step 2 is probably why the "evaluate my expressions without knowing anything about the underlying method" endpoint existed in the first place.